File: examples/go/escrow/Escrow_test.go
package contract

import (
	"testing"
	runar "github.com/icellan/runar/packages/runar-go"
)

func newEscrow() *Escrow {
	return &Escrow{
		Buyer:   runar.MockPubKey(),
		Seller:  runar.MockPubKey(),
		Arbiter: runar.MockPubKey(),
	}
}

func TestEscrow_ReleaseBySeller(t *testing.T)  { newEscrow().ReleaseBySeller(runar.MockSig()) }
func TestEscrow_ReleaseByArbiter(t *testing.T) { newEscrow().ReleaseByArbiter(runar.MockSig()) }
func TestEscrow_RefundToBuyer(t *testing.T)    { newEscrow().RefundToBuyer(runar.MockSig()) }
func TestEscrow_RefundByArbiter(t *testing.T)  { newEscrow().RefundByArbiter(runar.MockSig()) }

func TestEscrow_Compile(t *testing.T) {
	if err := runar.CompileCheck("Escrow.runar.go"); err != nil {
		t.Fatalf("Rúnar compile check failed: %v", err)
	}
}

File: examples/rust/escrow/Escrow_test.rs
#[path = "Escrow.tsop.rs"]
mod contract;

use contract::*;
use tsop::prelude::*;

fn new_escrow() -> Escrow {
    Escrow {
        buyer: mock_pub_key(),
        seller: mock_pub_key(),
        arbiter: mock_pub_key(),
    }
}

#[test] fn test_release_by_seller()  { new_escrow().release_by_seller(&mock_sig()); }
#[test] fn test_release_by_arbiter() { new_escrow().release_by_arbiter(&mock_sig()); }
#[test] fn test_refund_to_buyer()    { new_escrow().refund_to_buyer(&mock_sig()); }
#[test] fn test_refund_by_arbiter()  { new_escrow().refund_by_arbiter(&mock_sig()); }

#[test]
fn test_compile() {
    tsop::compile_check(include_str!("Escrow.tsop.rs"), "Escrow.tsop.rs").unwrap();
}
